packages/dcos-integration-test/extra/test_mesos.py

```diff
@@ -157,7 +158,12 @@ def test_if_ucr_app_runs_in_new_pid_namespace(dcos_api_session):
         marathon_framework_id = dcos_api_session.marathon.get('/v2/info').json()['frameworkId']
         app_task = dcos_api_session.marathon.get('/v2/apps/{}/tasks'.format(app['id'])).json()['tasks'][0]
 
-        content = dcos_api_session.mesos_sandbox_file(
-            app_task['slaveId'], marathon_framework_id, app_task['id'], ps_output_file)
+        # There is a short delay between the `app_task` starting and it writing
+        # its output to the `pd_output_file`. Because of this, we wait up to 10
+        # seconds for this file to appear before throwing an exception.
+        @retrying.retry(wait_fixed=0.1, stop_max_delay=10000, retry_on_exception=lambda x: True)
```

Comment:
Yikes, that's a major oversight. Thanks for catching it! My intention was to wait for 100ms or 0.1s. The file should appear almost immediately, so I only expect to have to wait between 0 and 100ms for it to show up (i.e. only error out once), but I can set it at 1s if you think that's preferable. I guess it depends on file system / network latencies.